lab_3/main.py

```diff
@@ -272,30 +508,57 @@ def register_language(self, language_profile: LanguageProfile) -> int:
         :param language_profile: a language profile
         :return: 0 if succeeds, 1 if not
         """
-        pass
+        if not isinstance(language_profile, LanguageProfile):
+            return 1
+        self.language_profiles[language_profile.language] = language_profile
+        return 0
 
-    def detect(self, unknown_profile: LanguageProfile, k: int, trie_levels: Tuple[int]) -> Dict[str, int] or int:
+    def detect(self, unknown_profile: LanguageProfile,
+               k: int, trie_levels: Tuple[int]) -> Dict[str, int] or int:
         """
         Detects the language of an unknown profile and its score
         :param unknown_profile: a dictionary
         :param k: a number of the most common n-grams
         :param trie_levels: N-gram size - tuple with one int for score 8
-        :return: a dictionary with language labels and their scores if input is correct, otherwise -1
+        :return: a dictionary with language labels
+        and their scores if input is correct, otherwise -1
         """
-        pass
+        if not isinstance(unknown_profile, LanguageProfile) or not isinstance(k, int) or \
+                not isinstance(trie_levels, Tuple):
+            return -1
+        return_dict = {}
+        for i in self.language_profiles.values():
+            distance = calculate_distance(unknown_profile, i, k, trie_levels[0])
+            return_dict[i.language] = distance
+        return return_dict
 
 
 def calculate_probability(unknown_profile: LanguageProfile, known_profile: LanguageProfile,
-                               k: int, trie_level: int) -> float or int:
+                          k: int, trie_level: int) -> float or int:
     """
-    Calculates probability of unknown_profile top_k ngrams in relation to known_profile
+    Calculates probability of unknown_profile top_k s in relation to known_profile
     :param unknown_profile: an instance of unknown profile
     :param known_profile: an instance of known profile
-    :param k: number of most frequent ngrams
-    :param trie_level: the size of ngrams
-    :return: a probability of unknown top k ngrams
+    :param k: number of most frequent s
+    :param trie_level: the size of s
+    :return: a probability of unknown top k s
     """
-    pass
+    if not isinstance(unknown_profile, LanguageProfile) \
+            or not isinstance(known_profile, LanguageProfile) \
+            or not isinstance(k, int) or not isinstance(trie_level, int):
+        return -1
+    known_trie = NGramTrie
```

Comment:
what is this assignment? It tells me that you assigned the whole class to one variable - what is the purpose?

Reply:
python swore that this variable might not be assigned and I just gave it a class)